src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1179,7 +1179,18 @@ public StringBuilder Append(char value)
                 return this;
             }
 
-            return Append(value.ToString());
+            return AppendSpanFormattableSlow(this, value);
+
+            static unsafe StringBuilder AppendSpanFormattableSlow<TValue>(StringBuilder sb, TValue value) where TValue : ISpanFormattable
+            {
+                const int bufferLength = 32; // should be enough for all primitives
```

Comment:
Should we assert TryFormat so the value does not get out of sync?

Reply:
The comment is still inaccurate.  It's not sufficient for all primitives in all cases because arbitrary culture settings could result in output being arbitrarily large.  It's good enough for the vast majority of cases,  though.